tests/test_scheduler/test_unit/test_base.py

```diff
@@ -0,0 +1,286 @@
+# pylint: disable=protected-access, unused-argument
+# pylint: disable=no-value-for-parameter
+
+import pytest
+import threading
+import radical.utils as ru
+from radical.pilot.agent.scheduler.base import AgentSchedulingComponent
+
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+
+# ------------------------------------------------------------------------------
+#
+@mock.patch.object(AgentSchedulingComponent, '__init__', return_value=None)
+def test_change_slot_states(mocked_init):
```

Comment:
I still think that test cases should be external json files.  What will this method look like in 5 years when we added 10 cases?   Who is going to maintain that when the function call or member var name changes?

The code could basically reduce to
```python
for test_case in glob('test_slots_*.json'):
    pre_cond, slots, post_cond = ru.read_json(test_case)
    component = AgentSchedulingComponent()
    component.nodes = pre_cond
    component._change_slot_states(slots=slots, new_state=1)
    assert(component.nodes == post_cond), test_case
```

with no loss of test coverage - but easy to maintain on code changes, and easy to add test cases: just drop a json file.